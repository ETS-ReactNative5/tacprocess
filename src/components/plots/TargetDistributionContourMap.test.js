import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TargetDistributionContourMap from "./TargetDistributionContourMap";

const targets = [
  {
    targetId: "Target: Target_Id 18378 Target_Id 18378 Name: 0, dtype: object",
    optional: true,
    ra: 0,
    dec: 0
  },
  {
    targetId: "Target: Target_Id 18378 Target_Id 18378 Name: 0, dtype: object",
    optional: true,
    ra: 55.363,
    dec: 489.331
  }
];

// Checking if the TargetDistributionContourMap Component renders correctly for different inputs
describe("TargetDistributionScatterPlot Component", () => {

  //Enzyme method testing if it renders correctly with empty value of the target
  it("Render the TargetDistributionContourMap Component having unpopulated props with no errors", () => {
    const rendered = mount(<TargetDistributionContourMap targets={[]}  />);
    expect(shallowToJson(rendered)).toMatchSnapshot();
  });

  it("Render the TargetDistributionContourMap Component having populated props with no errors", () => {
    const rendered = mount(<TargetDistributionContourMap targets={targets} />);
    expect(shallowToJson(rendered)).toMatchSnapshot();
  });
});
