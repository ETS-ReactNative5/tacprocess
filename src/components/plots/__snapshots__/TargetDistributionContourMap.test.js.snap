// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetDistributionScatterPlot Component Render the TargetDistributionContourMap Component having populated props with no errors 1`] = `
<TargetDistributionContourMap
  targets={
    Array [
      Object {
        "dec": 0,
        "optional": true,
        "ra": 0,
        "targetId": "Target: Target_Id 18378 Target_Id 18378 Name: 0, dtype: object",
      },
      Object {
        "dec": 489.331,
        "optional": true,
        "ra": 55.363,
        "targetId": "Target: Target_Id 18378 Target_Id 18378 Name: 0, dtype: object",
      },
    ]
  }
>
  <div>
    <h2>
      Target Distribution
    </h2>
    <svg
      className="plot"
      height={500}
      width={700}
    />
  </div>

</TargetDistributionContourMap>
`;

exports[`TargetDistributionScatterPlot Component Render the TargetDistributionContourMap Component having unpopulated props with no errors 1`] = `
<TargetDistributionContourMap
  targets={Array []}
>
  <div>
    <h2>
      Target Distribution
    </h2>
    <svg
      className="plot"
      height={500}
      width={700}
    />
  </div>

</TargetDistributionContourMap>
`;
